chapters/inheritance.tex

```diff
@@ -991,15 +991,21 @@ \subsection{Annotations for Redeclaration and Modification}\label{annotation-cho
 The annotation is not restricted to replaceable elements but can also be applied to non-replaceable elements, enumeration types, and simple variables.
 For a \lstinline!Boolean! variable, a \lstinline!choices! annotation may contain the definition \lstinline!checkBox = true!, meaning to display a checkbox to input the values \lstinline!false! or \lstinline!true! in the graphical user interface.
 
-The annotation \lstinline!choicesAllMatching = true!\annotationindex{choicesAllMatching} on a replaceable element indicates that tools should automatically construct a menu with choices of elements usable for replacing it.
+The annotation \lstinline!choicesAllMatching = true!\annotationindex{choicesAllMatching} on the following kinds of elements indicates that tools should automatically construct a menu with appropriate choices.
+\begin{itemize}
+\item For a replaceable element the included elements should be usable for replacing it.
 Exact criteria for inclusion in such a menu are not defined, but there shall be a a way to at least get a selection of classes, \lstinline!A.B.$\ldots$.X.Z!, that are either directly or indirectly derived by inheritance from the constraining class of the declaration, where \lstinline!A! to \lstinline!X! are non-partial packages, and \lstinline!Z! is non-partial.
+\item For a record variable the included elements shall include matching record constants and calls of matching record constructors (matching classes as for replaceable elements).
+\end{itemize}
+
 This menu can be disabled using annotation \lstinline!choicesAllMatching = false!.
 It is possible to combine the two annotations for one declaration, and tools may avoid generating duplicate menu entries in that case.
 \begin{nonnormative}
-The behavior when \lstinline!choicesAllMatching! is not specified; ideally it should present (at least) the same choices as for \lstinline!choicesAllMatching = true;! but if it takes (too long) time to present the list it is better to have \lstinline!choicesAllMatching = false!.
+The behavior when \lstinline!choicesAllMatching! is not specified; ideally it should for replaceable elements present (at least) the same choices as for \lstinline!choicesAllMatching = true;! but if it takes (too long) time to present the list it is better to have \lstinline!choicesAllMatching = false!.
```

Comment:
Since this sentence is being touched, we should take the opportunity to also make it more readable; the use of semicolon doesn't make sense to me after the first part.  Is that what is meant?
```suggestion
When \lstinline!choicesAllMatching! is not specified the following behavior is recommended.
For replaceable elements, a tool could ideally present (at least) the same choices as for \lstinline!choicesAllMatching = true!, but if it takes (too long) time to present the list it might be better to use the \lstinline!choicesAllMatching = false! behavior instead.
```